### DISCUSSION TITLE
Youtube-dl recent ITV "fixes"...

### DISCUSSION THREAD

**Discussion by Vangelis66** [Ascoware/get-iplayer-automator#299] [Ideas]
Latest `youtube-dl` (v**2020.12.12**) has come with two important **ITV** InfoExtractor (itv.py plugin) enhancements/fixes: 

1. Fix for https://github.com/ytdl-org/youtube-dl/issues/23177 :

[**itv**] Remove old extraction method (#23177)

2. Fix for https://github.com/ytdl-org/youtube-dl/issues/26897 : 

[**itv**] Fix series metadata extraction (#26897)

The patch responsible is: 

https://github.com/ytdl-org/youtube-dl/commit/225646c

@skovach, I think those would be beneficial to **GiA** (for ITV fetches), so when your spare time permits. the fix(es) should be incorporated into **GiA** code...

Many thanks, best festive wishes  😄 🎅

**Comment by skovatch** [comment]
I agree, that's actually a fairly substantial change. Thanks for the update!

**Comment by SOLOSAIR** [comment]
@Vangelis66 
>>I think those would be beneficial to GiA (for ITV fetches), so when your spare time permits. the fix(es) should be incorporated into GiA code...

ITV fetches are very much broken right now so I thank you *very much* indeed for providing this info! 👏🏼

**Comment by SOLOSAIR** [comment]
@skovatch Thank you for all the work you have been putting into updates 2021, Scott.  A few ITV progs are still halting just over half way through download, fans then going crazy.

Log for Vera 2021-01-04:

Get iPlayer Automator 1.20.1 Initialized.


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
GetITVShows: ITV Cache Update Starting 
<AppController: 0x7fa5c596e190>: Updating Programme Index Feeds...

INFO: Indexing tv programmes (concurrent)
GetITVShows (Info): Episodes: 2590 Today Programmes: 394 
GetITVShows: Update Finished
INFO: Added 4 tv programmes to cache
<AppController: 0x7fa5c596e190>: Index Updated.


INFO: Loading proxy settings...
INFO: No proxy to load
INFO: Proxy load complete.
INFO: No proxy will be used
AppController: Starting Downloads


Downloading Show 1/1:

ITV Download (ID=1a7314a0001): Downloading Vera - Mon Jan 04
INFO: Preparing Request for Auth Info
INFO: Metadata processed.
ITV Download (ID=1a7314a0001): youtube-dl finished downloading
AppController: Downloads Finished

---------

<img width="762" alt="Screenshot 2021-01-04 at 23 36 11" src="https://user-images.githubusercontent.com/37366634/103591168-43647100-4ee7-11eb-8ef3-e2b3adf67556.png">
<img width="354" alt="Screenshot 2021-01-04 at 23 36 25" src="https://user-images.githubusercontent.com/37366634/103591186-50816000-4ee7-11eb-9e89-5d72918ca8f0.png">

-----
Incidentally, when I tried stopping and restarting, I can't remember exactly what I did but this appeared and when I tried x-ing the windows, they all came back.  I had to force quit.

<img width="785" alt="Screenshot 2021-01-04 at 23 26 13" src="https://user-images.githubusercontent.com/37366634/103591474-2c724e80-4ee8-11eb-8220-d20c5ea96ff9.png">

Sorry if my posts annoy in any way, I'm just trying to help and am not tech savvy.
